platform/cluster.go

```diff
@@ -162,6 +162,17 @@ func (bc *BaseCluster) RenderUserData(userdata *conf.UserData, ignitionVars map[
 		conf.CopyKeys(keys)
 	}
 
+	if bc.OSContainer() != "" {
+		if bc.Distribution() != "rhcos" {
+			return nil, fmt.Errorf("oscontainer is only supported on the rhcos distribution")
+		}
+		conf.AddSystemdUnitDropin("pivot.service", "00-before-sshd.conf", `[Unit]
+Before=sshd.service`)
+		conf.AddSystemdUnit("pivot.service", "", true)
+		conf.AddSystemdUnit("pivot-write-reboot-needed.service", "[Unit]\nDescription=Touch /run/pivot/reboot-needed\nConditionFirstBoot=true\n\n[Service]\nType=oneshot\nExecStart=/usr/bin/mkdir /run/pivot\nExecStart=/usr/bin/touch /run/pivot/reboot-needed\n\n[Install]\nWantedBy=multi-user.target\n", true)
```

Comment:
A multi-line string would be clearer.

`mkdir -p` would be more robust.

Reply:
Will do, this unit was pulled from https://github.com/openshift/machine-config-operator/pull/363#issuecomment-463397373